app_flutter/test/commit_box_test.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:app_flutter/commit_box.dart';
+import 'package:mockito/mockito.dart';
+
+void main() {
+  group('CommitBox', () {
+    setUpAll(() async {
+      // Only needs to be done once since the HttpClient generated by this
+      // override is cached as a static singleton.
+      io.HttpOverrides.global = new TestHttpOverrides();
```

Comment:
I think this is starting to stray into the territory of testing `Image.network` instead of testing your widget.

A good guideline would be to look at the additional code coverage that code added to a test case gives you (use flutter test --coverage). If its not giving you more lines, you might be testing someone else's code.

Reply:
I wasn't happy with this approach, but otherwise the tests will fail as `Image.Network` will throw an exception for not being able to load the image.

I found [a StackOverflow on widgets tests with Image.Network](https://stackoverflow.com/a/52585057), but I figured we couldn't import that library to the project. So, I used [flutter/flutter-markdown](https://github.com/flutter/flutter_markdown/blob/a2fa53b7a29dc58b6f26b53e23a6f5a8d283dd66/test/flutter_markdown_test.dart#L506-L548) as the example on testing a widget with a network image.

Is there a better approach I should try?

Reply:
You can expect the failure, that should allow the test to pass, add this line to the end of the test:
```
tester.takeException();
```

Reply:
That worked. Thank you!